gapis/perfetto/android/trace.go

```diff
@@ -172,6 +172,15 @@ func Start(ctx context.Context, d adb.Device, a *android.ActivityAction, opts *s
 		}
 	}
 
+	// Enforce the AGI to launch the perfetto data producer, this will kill the
+	// running perfetto data producer and re-launch it if it's already running.
+	// And hence, any sequential update of the perfetto data producer or driver
+	// will likely lead to undefined behaviour.
```

Comment:
This comment is a bit confusing. It sounds like restarting (which is what we are doing) will cause undefined behavior. I'm assuming you are trying to say that restarting avoids the undefined behavior?

Reply:
ah. I am trying to say restarting might cause undefined behaviour. The scenario, which might be very rare, in my mind is:

1) A device is plugged in, device info is collected
2) the user starts tracing and in the same time, the user updates the driver package

Then AGI will start the data producer from the new package, where counters might be different. It shouldn't crash if I understand correctly, so I it's undefined.

Maybe I should say `any sequential update of the perfetto data producer or driver after device info is queried`

Reply:
Is the new comment better?

Reply:
merged your suggested comment. Thanks!